app/models/host/managed.rb

```diff
@@ -306,29 +306,6 @@ def pm_fqdn
     (puppetmaster == "puppet") ? "puppet.#{domain.name}" : puppetmaster.to_s
   end
 
-  # Cleans Certificate and enable Autosign
-  # Called before a host is given their provisioning template
-  # Returns : Boolean status of the operation
-  def handle_ca
-    # If there's no puppetca, tell the caller that everything is ok
-    return true unless Setting[:manage_puppetca]
-    return true unless puppetca?
-
-    # From here out, we expect things to work and return true
-    return false unless respond_to?(:initialize_puppetca, true)
-    return false unless initialize_puppetca
-    return false unless delCertificate
-
-    # If use_uuid_for_certificates is true, reuse the certname UUID value.
-    # If false, then reset the certname if it does not match the hostname.
-    if (Setting[:use_uuid_for_certificates] ? !Foreman.is_uuid?(certname) : certname != hostname)
-      logger.info "Removing certificate value #{certname} for host #{name}"
-      self.certname = nil
```

Comment:
Where do we handle this?

Reply:
When I looked at this, I wasn't sure when this would actually be relevant. `certname` is set depending on the `Setting[:use_uuid_for_certificates]`, so this would only be called when the setting is changed between creating the host and handling the ca. Since we are changing this to be at the same time, imho this should be a non issue.

Reply:
Found out we do need it, when the setting is changed between host creation and host connecting to PuppetCA, so we update the autosign entry. Re-Added this in the `queue_puppetca_update`.